### PR TITLE
[IntelliJ Plugin] Embed lsp plugin

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR provides direct language server support for ballerina IDEA plugin instead of connecting via the `LSP Support` dependency plugin.

### CODE REVIEW

File: tool-plugins/intellij/build.gradle [hunk 1]
```diff
@@ -40,11 +40,24 @@ repositories {
     maven { url 'http://dl.bintray.com/jetbrains/intellij-plugin-service' }
 }
 
-apply plugin: 'java'
+//LSP Dependencies
```
Comment (Shan1024) [inline tool-plugins/intellij/build.gradle]: Add a space after `//` and end with a full stop.

File: tool-plugins/intellij/build.gradle [hunk 2]
```diff
@@ -54,7 +67,7 @@ intellij {
     // For a full list of IntelliJ IDEA releases, please see https://www.jetbrains.com/intellij-repository/releases.
 
     version ideaVersion
-    plugins 'com.github.gtache.lsp:1.4.0'
+//    plugins 'com.github.gtache.lsp:1.4.0'
```
Comment (Shan1024) [inline tool-plugins/intellij/build.gradle]: Lets remove the commented codes.

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/completion/BallerinaKeywordContributor.java [hunk 3]
```diff
@@ -32,12 +31,8 @@
 public class BallerinaKeywordContributor extends CompletionContributor implements DumbAware {
 
     public BallerinaKeywordContributor() {
-        //If the LSP plugin is installed and enabled, disables the completions coming from ballerina plugin
-        if (!LSPluginInstallationNotificationProvider.isAlreadyInstalled() || LSPluginInstallationNotificationProvider
-                .isDisabled()) {
             extend(CompletionType.BASIC, isBallerinaNameReference(), new BallerinaKeywordCompletionProvider());
```
Comment (Shan1024) [inline tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/completion/BallerinaKeywordContributor.java]: Not formatted properly.

File: tool-plugins/intellij/src/main/resources/META-INF/plugin.xml [hunk 4]
```diff
@@ -200,15 +239,40 @@
                          shortName="BallerinaUnresolvedReference"
                          implementationClass="io.ballerina.plugins.idea.codeinspection.BallerinaUnresolvedReferenceInspection"/>
 
-        <editorNotificationProvider
-                implementation="io.ballerina.plugins.idea.inspections.LSPluginInstallationNotificationProvider"/>
+        <!--<editorNotificationProvider-->
+        <!--implementation="io.ballerina.plugins.idea.inspections.LSPluginInstallationNotificationProvider"/>-->
```
Comment (Shan1024) [inline tool-plugins/intellij/src/main/resources/META-INF/plugin.xml]: Lets remove this if not needed.

File: tool-plugins/intellij/src/main/resources/com/github/gtache/lsp/LSPBundle.properties [hunk 5]
```diff
@@ -0,0 +1 @@
+coursier.settings.textarea.basetext=Insert repositories like this (one per line):\nivy::https://...\nmaven::https://\nNote the two colons after the type.\nOnly maven and ivy repositories are supported at the moment.
```
Comment (Shan1024) [inline tool-plugins/intellij/src/main/resources/com/github/gtache/lsp/LSPBundle.properties]: Shall we add a newline here.

File: tool-plugins/intellij/build.gradle (hunk 1 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/build.gradle]: Fixed

File: tool-plugins/intellij/build.gradle (hunk 2 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/build.gradle]: Removed

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/completion/BallerinaKeywordContributor.java (hunk 3 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/completion/BallerinaKeywordContributor.java]: Formatted

File: tool-plugins/intellij/src/main/resources/META-INF/plugin.xml (hunk 4 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/src/main/resources/META-INF/plugin.xml]: Removed

File: tool-plugins/intellij/build.gradle [hunk 6]
```diff
@@ -40,11 +40,24 @@ repositories {
...
 }
 
-apply plugin: 'java'
+// LSP Dependencies
+dependencies {
+    compile group: 'org.eclipse.lsp4j', name: 'org.eclipse.lsp4j', version: '0.4.1'
+    compile group: 'io.get-coursier', name: 'coursier_2.12', version: '1.0.3'
+    compile group: 'io.get-coursier', name: 'coursier-cache_2.12', version: '1.0.3'
+    compile group: 'com.vladsch.flexmark', name: 'flexmark', version: '0.32.24'
+    compile 'org.scala-lang:scala-compiler:2.12.4'
```
Comment (Shan1024) [inline tool-plugins/intellij/build.gradle]: Is it possible to uniform these dependency declarations with the above ones?

Comment (NipunaRanasinghe) [reply tool-plugins/intellij/build.gradle]: Yes it is, done

**Pull request merged**